### PR TITLE
document fields of SwapInfo

### PR DESCRIPTION
**Pull request by ubbabeck**
documents #537

### CODE REVIEW

File: libs/sdk-core/src/models.rs [hunk 1]
```diff
@@ -1169,32 +1169,56 @@ impl TryFrom<i32> for SwapStatus {
...
     pub paid_sats: u64,
+    /// Confirmed onchain sats to be claim with an bolt11 invoice or refunded if swap fails.
     pub confirmed_sats: u64,
+    /// Unconfirmed sats waiting to be confirmed onchain.
     pub unconfirmed_sats: u64,
+    /// Shows the current status of the swap, either `Initial` or `Expired`.
     pub status: SwapStatus,
+    /// Transaction ID for failed swap attempts.
     pub refund_tx_ids: Vec<String>,
+    /// Transaction ID for ongoing swap awaiting confirmation.
```
Comment (ok300) [inline libs/sdk-core/src/models.rs:1211]:
```suggestion
    /// Transaction IDs for ongoing swap awaiting confirmation.
```

File: libs/sdk-core/src/models.rs [hunk 2]
```diff
@@ -1169,32 +1169,56 @@ impl TryFrom<i32> for SwapStatus {
...
     pub unconfirmed_tx_ids: Vec<String>,
+    /// Transaction ID that have been confirmed on-chain.
```
Comment (ok300) [inline libs/sdk-core/src/models.rs:1213]:
```suggestion
    /// Transaction IDs that have been confirmed on-chain.
```

File: libs/sdk-core/src/models.rs [hunk 3]
```diff
@@ -1169,32 +1169,56 @@ impl TryFrom<i32> for SwapStatus {
...
     pub confirmed_tx_ids: Vec<String>,
+    /// The minimum amount the one can send in order for the swap to succeed. Received from [SwapperAPI::create_swap].
```
Comment (ok300) [inline libs/sdk-core/src/models.rs:1215]:
```suggestion
    /// The minimum amount of sats one can send in order for the swap to succeed. Received from [SwapperAPI::create_swap].
```

File: libs/sdk-core/src/models.rs [hunk 4]
```diff
@@ -1169,32 +1169,56 @@ impl TryFrom<i32> for SwapStatus {
...
     pub min_allowed_deposit: i64,
+    /// The maximum amount the one can send in order for the swap to succeed. Received from [SwapperAPI::create_swap].
```
Comment (ok300) [inline libs/sdk-core/src/models.rs:1217]:
```suggestion
    /// The maximum amount of sats one can send in order for the swap to succeed. Received from [SwapperAPI::create_swap].
```

**Review by ok300 — COMMENTED** [review]
A few NITs, but generally looks good!

File: libs/sdk-core/src/models.rs [hunk 5]
```diff
@@ -1170,51 +1170,48 @@ impl TryFrom<i32> for SwapStatus {
...
+    /// Amount of sats payed to the claim address. Dynamic field
     pub paid_sats: u64,
-    /// Confirmed onchain sats to be claim with an bolt11 invoice or refunded if swap fails.
+    /// Confirmed onchain sats to be claim with an bolt11 invoice or refunded if swap fails. Dynamic field.
     pub confirmed_sats: u64,
-    /// Unconfirmed sats waiting to be confirmed onchain.
+    /// Unconfirmed sats waiting to be confirmed onchain. Dynamic field.
     pub unconfirmed_sats: u64,
-    /// Shows the current status of the swap, either `Initial` or `Expired`.
+    /// Shows the current status of the swap, either `Initial` or `Expired`. Dynamic field. Dynamic field.
```
Comment (ok300) [inline libs/sdk-core/src/models.rs:1204]:
```suggestion
    /// Shows the current status of the swap, either `Initial` or `Expired`. Dynamic field.
```

File: libs/sdk-core/src/models.rs [hunk 6]
```diff
@@ -1169,32 +1169,56 @@ impl TryFrom<i32> for SwapStatus {
 /// saved to the persistent storage.
 ///
 /// The SwapInfo has a status which changes accordingly, documented in [SwapStatus].
+///
+/// Static imutable fields are [`bitcoin_address`](SwapInfo::bitcoin_address), [`created_at`](SwapInfo::created_at), [`lock_height`](SwapInfo::lock_height), [`payment_hash`](SwapInfo::payment_hash), [`preimage`](SwapInfo::preimage), [`private_key`](SwapInfo::private_key), [`public_key`](SwapInfo::public_key), [`swapper_public_key`](SwapInfo::swapper_public_key), [`script`](SwapInfo::script), [`min_allowed_deposit`](SwapInfo::min_allowed_deposit) and [`max_allowed_deposit`](SwapInfo::max_allowed_deposit).
```
[the remark this answers is not in the document]
Comment (ubbabeck) [reply libs/sdk-core/src/models.rs]: Good point, added the short description and whenever the fields are dynamic or static

File: libs/sdk-core/src/models.rs (hunk 5 above)
Comment (ubbabeck) [reply libs/sdk-core/src/models.rs]: Whoops, should have seen that. Thanks.

**Review by ok300 — APPROVED** [review]
LGTM

File: libs/sdk-core/src/models.rs [hunk 7]
```diff
@@ -1169,32 +1169,53 @@ impl TryFrom<i32> for SwapStatus {
 /// saved to the persistent storage.
 ///
 /// The SwapInfo has a status which changes accordingly, documented in [SwapStatus].
+///
+/// Static fields are those which don't change for the lifetime of a swap, whereas dynamic fields may change.
+
 #[derive(Clone, PartialEq, Eq, Debug, Serialize, Deserialize)]
 pub struct SwapInfo {
-    //static immutable data
+    /// Bitcoin address for this swap. Sats sent to this address will be swapped. Static field.
```
Comment (roeierez) [inline libs/sdk-core/src/models.rs:1177]: Static field doesn't seem the right term. Perhaps "immutable field" more fits here although I don't feel we should add that info.

File: libs/sdk-core/src/models.rs [hunk 8]
```diff
@@ -1169,32 +1169,53 @@ impl TryFrom<i32> for SwapStatus {
...
     pub public_key: Vec<u8>,
+    /// The public key in binary format from the swapping service. Received from [SwapperAPI::create_swap]. Static field.
     pub swapper_public_key: Vec<u8>,
+    /// The lockingsscript for the generated bitcoin address. Received from [SwapperAPI::create_swap]. Static field.
     pub script: Vec<u8>,
 
-    // dynamic data
+    /// bolt11 invoice to claim the sent funds. Dynamic field.
     pub bolt11: Option<String>,
+    /// Amount of sats payed to the claim address. Dynamic field
```
Comment (roeierez) [inline libs/sdk-core/src/models.rs:1198]: Same here for Dynamic field. I would remove this, too low level and implementation details.

**Review by roeierez — CHANGES_REQUESTED** [review]

File: libs/sdk-core/src/models.rs (hunk 7 above)
Comment (ubbabeck) [reply libs/sdk-core/src/models.rs]: Good point, might be needless to add this info. Removed

File: libs/sdk-core/src/models.rs (hunk 8 above)
Comment (ubbabeck) [reply libs/sdk-core/src/models.rs]: Done

File: libs/sdk-core/src/models.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (ubbabeck) [reply libs/sdk-core/src/models.rs]: Removed not longer relevant ref (https://github.com/breez/breez-sdk/pull/712#discussion_r1449926815)

File: libs/sdk-core/src/models.rs [hunk 9]
```diff
@@ -1169,32 +1169,52 @@ impl TryFrom<i32> for SwapStatus {
...
+    /// Transaction IDs that have been confirmed on-chain.
     pub confirmed_tx_ids: Vec<String>,
+    /// The minimum amount of sats one can send in order for the swap to succeed. Received from [SwapperAPI::create_swap].   
     pub min_allowed_deposit: i64,
+    /// The maximum amount of sats one can send in order for the swap to succeed. Received from [SwapperAPI::create_swap].
     pub max_allowed_deposit: i64,
+    /// Error reason for when swap fails.
     pub last_redeem_error: Option<String>,
-    /// The dynamic fees that may have been used when the swap was created
+    /// The dynamic fees which is present if a channel opening is needed.
```
Comment (roeierez) [inline libs/sdk-core/src/models.rs:1217]: "which is present" => "Which is set" ?

File: libs/sdk-flutter/lib/bridge_generated.dart [hunk 10]
```diff
@@ -1672,29 +1672,69 @@ sealed class SuccessActionProcessed with _$SuccessActionProcessed {
...
   final int minAllowedDeposit;
+
+  /// The maximum amount of sats one can send in order for the swap to succeed. Received from [SwapperAPI::create_swap].
   final int maxAllowedDeposit;
+
+  /// Error reason for when swap fails.
   final String? lastRedeemError;
 
-  /// The dynamic fees that may have been used when the swap was created
+  /// The dynamic fees which is present if a channel opening is needed.
```
Comment (roeierez) [inline libs/sdk-flutter/lib/bridge_generated.dart:1737]: "Which is set" ?

**Review by roeierez — CHANGES_REQUESTED** [review]
Small two syntax changes. Other than that Looks good.

File: libs/sdk-flutter/lib/bridge_generated.dart (hunk 10 above)
Comment (ubbabeck) [reply libs/sdk-flutter/lib/bridge_generated.dart]: Added.

File: libs/sdk-core/src/models.rs (hunk 9 above)
Comment (ubbabeck) [reply libs/sdk-core/src/models.rs]: Added

**Pull request merged**